test/test_library/unit/test_init.py

```diff
@@ -0,0 +1,1858 @@
+
+        # testing
+        ident, location = lib.locate(ref)
+        self.assertIsNone(ident)
+        self.assertIsNone(location)
+
+    def test_returns_ident_and_cache_if_ident_is_cached(self):
+        # prepare state
+        lib = Library(self.cache, self.sqlite_db)
+
+        fake_cache = fudge.Fake('cache').expects('has').returns(True)
+        lib.cache = fake_cache
+
+        ds1 = DatasetFactory()
+
+        # testing
+        ret1, ret2 = lib.locate(ds1.identity)
+        fudge.verify()
+        self.assertIsInstance(ret1, Identity)
+        self.assertEquals(ret1.vid, ds1.vid)
+        self.assertEquals(ret2, fake_cache)
+
+    def test_returns_ident_and_remote_location_if_ident_is_not_cached(self):
+        # prepare state
+        lib = Library(self.cache, self.sqlite_db)
+        lib.cache = fudge.Fake('cache').expects('has').returns(False)
+
+        # also, remote have to contain ident.
+        remote1 = fudge.Fake()\
+            .has_attr(priority=1)\
+            .expects('has')\
+            .returns(True)
+
+        lib._remotes = {
+            'remote1': remote1}
+        ds1 = DatasetFactory()
+
+        # testing
+        ret1, ret2 = lib.locate(ds1.identity)
+        fudge.verify()
+        self.assertIsInstance(ret1, Identity)
+        self.assertEquals(ret1.vid, ds1.vid)
+        self.assertEquals(ret2, remote1)
+
+    def test_returns_remote_location_with_lowest_priority(self):
+        print('\nTODO: what is the reason about low priority?\n')
+
+        # prepare state
+        lib = Library(self.cache, self.sqlite_db)
+        lib.cache = fudge.Fake('cache').expects('has').returns(False)
+
+        # also, remote have to contain ident.
+        remote1 = fudge.Fake()\
+            .has_attr(priority=5)
+
+        remote2 = fudge.Fake()\
+            .has_attr(priority=3)
+
+        remote3 = fudge.Fake()\
+            .has_attr(priority=1)\
+            .expects('has')\
+            .returns(True)
+
+        lib._remotes = {
+            'remote1': remote1,
+            'remote2': remote2,
+            'remote3': remote3}
+        ds1 = DatasetFactory()
+
+        # testing
+        ret1, ret2 = lib.locate(ds1.identity)
+        fudge.verify()
+        self.assertIsInstance(ret1, Identity)
+        self.assertEquals(ret1.vid, ds1.vid)
+        self.assertEquals(ret2, remote3)
+
+    def test_returns_ident_and_none_if_location_not_found(self):
+
+        # prepare state
+        lib = Library(self.cache, self.sqlite_db)
+        lib.cache = fudge.Fake('cache').expects('has').returns(False)
+        lib._remotes = {}
+        ds1 = DatasetFactory()
+
+        # testing
+        ret1, ret2 = lib.locate(ds1.identity)
+        fudge.verify()
+        self.assertIsInstance(ret1, Identity)
+        self.assertEquals(ret1.vid, ds1.vid)
+        self.assertIsNone(ret2)
+
+    # .locate_one test
+    # @unittest.skip('.pop() on tuple returned by locate()? Is it highest or lowest?')
```

Comment:
I'm talking about another thing. Look at Library.locate method - https://github.com/nmb10/ambry/blob/issue-8/ambry/library/__init__.py#L836. It returns tuple
``` python
return ident, None
```
But Library.locate_one expects something with pop() method - https://github.com/nmb10/ambry/blob/issue-8/ambry/library/__init__.py#L862 
```python
return f.pop(0)
```
One of them should be changed. But which one?

P.S. I didn't find any call to locate_one. I think deleting unused code is good idea. What do you think?